src/interfaces/token.spec.ts

```diff
@@ -590,297 +589,97 @@ describe("TokenInterface", () => {
   });
 
   describe("approve", () => {
-    describe("when the vault token is the same as the token", () => {
-      let vault: Asset<"VAULT_V2">;
-      let token: Address;
+    const owner: Address = "0xOwner";
+    const spender: Address = "0xSpender";
+    const amount: Integer = "1000000";
+    let token: Address;
 
-      beforeEach(() => {
-        vault = createMockAssetStaticVaultV2();
-        token = createMockToken().address;
-        sendTransactionMock.mockResolvedValue(true);
-      });
-
-      describe("when it has a partner id defined", () => {
-        it("should approve the partner contract to spend a token on a direct deposit", async () => {
-          mockedYearn = new (Yearn as jest.Mock<Yearn<ChainId>>)();
-          mockedYearn.services.partner = new ((PartnerService as unknown) as jest.Mock<PartnerService<ChainId>>)();
-          tokenInterface = new TokenInterface(mockedYearn, 1, new Context({}));
-          approveMock.mockReturnValue("partner deposit");
-          const actualApprove = await tokenInterface.approveDeposit(vault.address, vault.token, token, "1", "0x001");
-
-          expect(actualApprove).toEqual(true);
-          expect(Contract).toHaveBeenCalledTimes(1);
-          expect(Contract).toHaveBeenCalledWith(
-            token,
-            [
-              "function approve(address _spender, uint256 _value) public",
-              "function allowance(address _owner, address _spender) public view returns (uint256)"
-            ],
-            {
-              sendTransaction: expect.any(Function)
-            }
-          );
-          expect(approveMock).toHaveBeenCalledTimes(1);
-          expect(approveMock).toHaveBeenCalledWith("0x0001partner", "1");
-          expect(sendTransactionMock).toHaveBeenCalledTimes(1);
-          expect(sendTransactionMock).toHaveBeenCalledWith("partner deposit");
-        });
-      });
-
-      it("should approve vault to spend a token on a direct deposit", async () => {
-        approveMock.mockReturnValue("direct deposit");
-        const actualApprove = await tokenInterface.approveDeposit(vault.address, vault.token, token, "1", "0x001");
-
-        expect(actualApprove).toEqual(true);
-
-        expect(Contract).toHaveBeenCalledTimes(1);
-        expect(Contract).toHaveBeenCalledWith(
-          token,
-          [
-            "function approve(address _spender, uint256 _value) public",
-            "function allowance(address _owner, address _spender) public view returns (uint256)"
-          ],
-          {
-            sendTransaction: expect.any(Function)
-          }
-        );
-        expect(approveMock).toHaveBeenCalledTimes(1);
-        expect(approveMock).toHaveBeenCalledWith(vault.address, "1");
-        expect(sendTransactionMock).toHaveBeenCalledTimes(1);
-        expect(sendTransactionMock).toHaveBeenCalledWith("direct deposit");
-      });
+    beforeEach(() => {
+      token = createMockToken().address;
```

Comment:
This feel misleading, if we just want the address I would do something like:
```suggestion
      const { address } = createMockToken();
```
Otherwise would rename that variable to `tokenAddress`, as it's not a token as the name implies

Reply:
moved to constants and typed it as `Address` here https://github.com/yearn/yearn-sdk/pull/266/commits/bb5bb305da7cda5bbb9a7763986886dafadf5f19

Reply:
The problem is that that type is not exposed when you hover the variable, so it will just say `string`

I'd be a bit more explicit in the naming

Reply:
Ended up changing it here https://github.com/yearn/yearn-sdk/commit/5949fd4abc1094090e6bdd0f97a70bf07ee59604 after reading the next comment